pkg/common/fileutils/fileutils.go

```diff
@@ -150,6 +150,34 @@ func GetAllFilesWithSameNameInDirectory(dir string, fname string) []string {
 	return result
 }
 
+func GetAllFilesWithSpecificExtensions[T any](dir string, ext map[string]T) []string {
```

Comment:
nit: It looks like the value `T` is never used. Is there a reason you're using `map[string]T` instead of `map[string]struct{}`?

Reply:
Just used T since, the caller's doesn't have to be send this as a specific type. I just made it for generalization. But yeah can change it to `map[string]struct{}`